.github/workflows/docs.yml

```diff
@@ -0,0 +1,31 @@
+name: Docs
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  build-and-deploy-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install doxygen sphinx-doc -y
+          pip3 install sphinx-rtd-theme breathe sphinx-sitemap m2r2
+          pip3 uninstall mistune -y
+          pip3 install mistune==0.8.4
+
+      - name: Generate docs
+        run: |
+          cd docs
+          make html
+          cd _build/html
+          touch .nojekyll
+
+      - name: Deploy to GitHub pages
+        uses: JamesIves/github-pages-deploy-action@4.1.5
+        with:
+          TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
@Benzammour I think that you might need to do this as an owner. The is a description here: https://github.com/marketplace/actions/deploy-to-github-pages